qiskit_ibm_runtime/sampler.py

```diff
@@ -90,11 +91,36 @@ def __init__(
         Sampler.__init__(self)
         BasePrimitiveV2.__init__(self, backend=backend, session=session, options=options)
 
-        raise NotImplementedError("SamplerV2 is not currently supported.")
-
         # if self._service._channel_strategy == "q-ctrl":
         #     raise NotImplementedError("SamplerV2 is not supported with q-ctrl channel strategy.")
 
+    def run(self, pubs: SamplerPubLike | Iterable[SamplerPubLike]) -> RuntimeJob:
+        """Submit a request to the sampler primitive.
+
+        Args:
+            pubs: A Primitive Unified Bloc or a list of such blocs
+
+        Returns:
+            Submitted job.
+            The result of the job is an instance of
+            :class:`qiskit.primitives.containers.PrimitiveResult`.
+
+        Raises:
+            ValueError: Invalid arguments are given.
+        """
+        pubs_to_run: List[SamplerPub]
+        if isinstance(pubs, SamplerPub):
+            pubs_to_run = [pubs]
+        elif isinstance(pubs, tuple) and isinstance(pubs[0], QuantumCircuit):
+            pubs_to_run = [SamplerPub.coerce(pubs)]
+        elif pubs is not SamplerPub:
+            pubs_to_run = [SamplerPub.coerce(pub) for pub in pubs]
+
+        for pub in pubs_to_run:
+            pub.validate()
```

Comment:
I removed the validation block because `coerce` does validation (except when the input is already a SamplerPub, but in this case the user would have had to turn off validation manually because the constructor does it, so that's on them).

Note that we _don`t_ need to pass the `shots` kwarg to `_run` because coersion has resolved the values and placed them into each pub, which takes precedence.

```suggestion
        pubs_to_run: List[SamplerPub]
        if isinstance(pubs, SamplerPub):
            pubs_to_run = [pubs]
        elif isinstance(pubs, tuple) and isinstance(pubs[0], QuantumCircuit):
            pubs_to_run = [SamplerPub.coerce(pubs, shots)]
        elif pubs is not SamplerPub:
            pubs_to_run = [SamplerPub.coerce(pub, shots) for pub in pubs]
```